moq-arch.md

```diff
@@ -2,87 +2,99 @@
-
-The Relays are arranged in a logical tree where, for a given application,
-there is an origin Relay at root of the tree that controls the
-namespace. Publish messages are sent towards the root of the tree and
-down the path of any subscribers to that named data.
-
-The QuicR protocol takes care of transmitting named objects from the
-Publisher to the Relay and from the Relay to all the subscribers of the
-named object. It provides transport services selected and tuned based on
-application requirements (with the support of underlying transport,
-where necessary) such as detecting available bandwidth, fragmentation
-and reassembly, resiliency, congestion control and prioritization of
+what CDN provides today. The subscribe messages allow subscription to a 
+name that includes a wildcard to match multiple published names, so a 
+single subscribe can allow a client to receive publishes for a wide class 
+of named objects. Objects are named such that it is unique for the
+relay/delivery network and scoped to an application. 
+
+QuicrR provides services based on application requirements 
+(with the support of underlying transport, where necessary) 
+such as detecting available bandwidth, fragmentation and 
+reassembly, resiliency, congestion control and prioritization of
 data delivery based on data lifetime and importance of data. It is
 designed to be NAT and firewall traversal friendly and can be fronted
-with load balancers. Objects are named such that it is unique for the
-relay/delivery network and scoped to an application. Subscriptions can
-include a form of wildcarding to the named object.
+with load balancers. 
+
+The Relays are arranged in a logical tree (as shown below) where, 
+for a given application, there is an origin Relay at root of the tree 
+that controls the namespace. Publish messages are sent towards the 
+root of the tree and down the path of any subscribers to that 
+named object. QuicR is designed to make it easy to implement relays 
+so that fail over could happen between relays with minimal impact to 
+the clients and relays can redirect a client to a different relay.
+
+!--
+~~~ascii-art 
+              ┌────────────┐
+              │            │
+              │            ▼
+              │       ┌────────┐
+              │   ▬ ▬▶│Relay-0 │ ◀▬▬ ▬▬ ▬▮
+          pub │  ▮    │ Orgin  ├┐        ▮
+              │  ▮     ────────┘│        ▮
+              │  ▮ sub          │        ▮ sub
+              │  ▮          pub │        ▮
+              │  ▮              │        ▮
+         ┌───────▮┐ ◀▬▮         │  ┌─────▮──┐
+     ┌──▶│ Relay-1│   ▮         └─▶│ Relay-2│◀▮▮
+     │   └─────┬──┘   ▮             ▲───┬───┘  ▮
+ pub │         │      ▮ sub     sub ▮   │      ▮ sub
+     │      pub│      ▮             ▮   │      ▮
+    ┌┴─────┐   │ ┌────▮─┐     ┌─────▮┐  │  ┌───▮──┐
+    │Alice │   └▶│ Bob  │     │ Carl │◀─┴─▶│Derek │
+    └──────┘     └──────┘     └──────┘ pub └──────┘
```

Comment:
I think the pub line from Relay 2 to Carl and Derek would be better we could show it as two lines. Love the bold lines. What tool did you use for this ?

Reply:
Fixed